whois.go

```diff
@@ -49,21 +52,44 @@ func License() string {
 	return "Licensed under the Apache License 2.0"
 }
 
+type Client struct {
+	dialer  proxy.Dialer
+	timeout time.Duration
+}
+
+func NewClient(dialer proxy.Dialer, timeout *time.Duration) (*Client, error) {
```

Comment:
`proxy dialer` may be too heavy here.
Will it more clear if set a proxy after `NewClient`?